policies/coding-style.md

```diff
@@ -234,27 +234,44 @@ changing their context lines.
 
 ## Chapter 4: Naming
 
-C is a Spartan language, and so should your naming be. Do not use long
-names like _ThisVariableIsATemporaryCounter_. Use a name like _tmp_, which
-is much easier to write, and not more difficult to understand.
+C is a Spartan language, and so should your naming be.
 
-Except when otherwise required, avoid mixed-case names.
+Local variable names should be short, and to the point. If you have
+some random integer loop counter, it should probably be called `i` or `j`.
+Apart from that, avoid single-letter names,
+in particular when they can be visually confusing, such as `I` and `O`.
+Calling a varable `loop_counter` is non-productive, if there is no chance of it
+being mis-understood. Similarly, `tmp` can be just about any type of
+variable that is used to hold a temporary value.
```

Comment:
I do not understand the purpose of the sentence with `tmp` - is that recommending to use `tmp` or against that?

Reply:
I don't know, either.
This sentence is part of the original text - it appears 'new' just because I re-arranged the order of the paragraphs.

Reply:
I'd remove it altogether.

Reply:
Hmm, actually the original paragraph was fine and I could understand as supporting `tmp` for a variable name if there is no ambiguity, i.e. only one temporary variable in the function.

Reply:
I would even say the original paragraph was just fine and if you just moved it, it would be OK. Because the sentence about `loop_counter` goes well with the previous sentence in the original wording, not so much in your edited version.

Reply:
Note that previously there were two paragraphs mentioning the name `tmp`:
```
Do not use long
names like _ThisVariableIsATemporaryCounter_. Use a name like _tmp_, which
is much easier to write, and not more difficult to understand.
```
I removed the this one because it was redundant,
and this is the one you found more easy to understand, and I agree on this.

The second one is this:
```
Local variable names should be short, and to the point. If you have
some random integer loop counter, it should probably be called _i_.
Calling it _loop_counter_ is non-productive, if there is no chance of it
being mis-understood. Similarly, _tmp_ can be just about any type of
variable that is used to hold a temporary value
```
and this is the one I just moved around.

Unfortunately, its last sentence is hard to understand.
How about replacing it by, e.g.:
```
Use brief name name like `tmp` and `name` as long as they are non-ambiguous in the given context.
```

Reply:
> How about replacing it by, e.g.:
> 
> ```
> Use brief name name like `tmp` and `name` as long as they are non-ambiguous in the given context.
> ```

That would be fine.

Reply:
Done